storage/factory/pruningStorerFactory.go

```diff
@@ -366,21 +364,33 @@ func (psf *StorageServiceFactory) CreateForMeta() (dataRetriever.StorageService,
 	store.AddStorer(dataRetriever.StatusMetricsUnit, statusMetricsStorageUnit)
 	store.AddStorer(dataRetriever.TxLogsUnit, txLogsUnit)
 
-	if psf.generalConfig.FullHistory.EnableHistoryNode {
-		historyTxsUnitArgs := psf.createPruningStorerArgs(psf.generalConfig.FullHistory.HistoryTransactionStorageConfig)
-		historyTxUnit, err := pruning.NewPruningStorer(historyTxsUnitArgs)
-		if err != nil {
-			return nil, err
-		}
+	historyTxUnit, err := psf.createHistoryStorerIfNeeded()
+	if err != nil {
+		return nil, err
+	}
 
+	if psf.generalConfig.FullHistory.EnableHistoryNode {
```

Comment:
Please move also the if clause with addStorer to the createHistoryStorerIfNeeded function, you can give the store as parameter.

Reply:
yeah i can give store parameter but i have a problem with `successfullyCreatedStorers` because  i have to return if storer was created or not to know if I have to add it into slice.

Reply:
Perhaps, if nil - don't `AddStorer`?